server/configuration.go

```diff
@@ -155,12 +156,25 @@ func (p *Plugin) diffConfiguration(newConfiguration *configuration) {
 			continue
 		}
 
+		newConfigurationData, jsonErr := json.Marshal(newConfiguration)
+		if jsonErr != nil {
+			p.API.LogWarn("failed to marshal new configuration", "err", err)
+			return
+		}
+
+		fileInfo, err := p.API.UploadFile(newConfigurationData, demoChannelId, "configuration.json")
```

Comment:
are files tied to a user, and only that user is allowed to attach a file with a post or are files tied to a channel and anybody in that channel can tie that file to a post?

Reply:
Files may be attached to a user (via the `CreatorId`) or to `nouser`. They are always bound to a channel (not 100% certain why!). Files bound to a specific user may only be used by that user. Files bound to `nouser` may be used by anyone. However, once a file is bound to a post, it cannot be rebound.